src/main/java/com/spotify/github/v3/clients/RepositoryClient.java

```diff
@@ -281,6 +307,112 @@ public ChecksClient createChecksApiClient() {
     return github.request(path, Commit.class);
   }
 
+  /**
+   * Get  repository commit and map to CommitResponse.
+   *
+   * @param sha commit sha
+   * @return commit response
+   */
+  public CompletableFuture<CommitResponse> getGitCommit(final String sha) {
+    final String path = String.format(COMMIT_SHA_URI_TEMPLATE, owner, repo, sha);
+    return github.request(path, CommitResponse.class);
+  }
+
+  /**
+   * Post new content to the server.
+   *
+   * @param content the content to be posted
+   */
+  public CompletableFuture<Response> setBlob(final String content) {
+    final String path = String.format(BLOB_TEMPLATE, owner, repo);
+    final String encoding = "utf-8|base64";
+    final String requestBody = github.json()
+        .toJsonUnchecked(ImmutableMap.of("content", content, "encoding", encoding));
+    return github.post(path, requestBody);
+  }
+
+  /**
+   * Create a commit which references a tree
+   *
+   * @param message commit message
+   * @param parents list of parent sha values, usually just one sha
+   * @param treeSha sha value of the tree
+   */
+  public CompletableFuture<Response> setCommit(final String message, final List<String> parents,
+      final String treeSha) {
+    final String path = String.format(COMMIT_TEMPLATE, owner, repo);
+    final String requestBody = github.json()
+        .toJsonUnchecked(ImmutableMap.of("message", message, "parents", parents, "tree", treeSha));
+    return github.post(path, requestBody);
+  }
+
+  /**
+   * Create a commit which references a tree.
+   *
+   * @param content new content.
+   * @param ref     reference to base branch for the commit.
+   * @param branch  name of new branch, must start with refs/heads.
+   * @param path    to the file the changes should be applied to.
+   * @param message the commit message
+   */
+  public CompletableFuture<Response> createCommit(final String content, final String ref,
+      final String branch, final String path, final String message) {
+
+    CommitWrapper commitWrapper = new CommitWrapper();
+    Wrapper blobWrapper = new Wrapper();
+
+    return getReference(ref).thenCompose(
+            referenceResponse -> getGitCommit(referenceResponse.object().sha()))
+        .thenCompose(commitResponse -> {
+          commitWrapper.setSha(commitResponse.sha());
+          commitWrapper.setTreeSha(commitResponse.commit().tree().sha());
+          commitWrapper.setTreeUrl(commitResponse.commit().tree().url());
+          return setBlob(content);
+        }).thenCompose(blobResponse -> {
+          try {
+            assert blobResponse.body() != null;
+            String sha = Json.create().fromJson(blobResponse.body().string(), ShaLink.class).sha();
+            blobWrapper.setSha(sha);
+            return getTree(commitWrapper.getTreeSha());
+          } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
It would be better to throw a more specific exception here rather then a generic `RuntimeException`. For example there `GithubException` in the repo, you can use that: https://github.com/spotify/github-java-client/blob/master/src/main/java/com/spotify/github/v3/exceptions/GithubException.java